src/Microsoft.AspNetCore.Mvc.Core/ControllerContext.cs

```diff
@@ -52,18 +52,18 @@ public new ControllerActionDescriptor ActionDescriptor
         }
 
         /// <summary>
-        /// Gets or sets the list of <see cref="IValueProvider"/> instances for the current request.
+        /// Gets or sets the list of <see cref="IValueProviderFactory"/> instances for the current request.
         /// </summary>
-        public virtual IList<IValueProvider> ValueProviders
+        public virtual IList<IValueProviderFactory> ValueProviderFactories
```

Comment:
We intend for this to be changed in scenarios where you want to prevent MVC model binding from reading the form data (large file uploads). We had a meeting about this and everything.